From 275feae3a76be38b3b9ed799d9b614fed26ef206 Mon Sep 17 00:00:00 2001
From: Marko Katic <dromede.gmail.com>
Date: Sat, 7 Jan 2012 19:46:55 +0100
Subject: [PATCH 10/15] Force mousedev to ignore the ads7846 touchscreen. I does so by comparing
 vendor and id tags of the device. There might be corner cases where it does not
 work.

---
 drivers/input/mousedev.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/drivers/input/mousedev.c b/drivers/input/mousedev.c
index 0110b5a..6c8c965 100644
--- a/drivers/input/mousedev.c
+++ b/drivers/input/mousedev.c
@@ -972,7 +972,12 @@ static int mousedev_connect(struct input_handler *handler,
 	struct mousedev *mousedev;
 	int minor;
 	int error;
-
+	
+	if (dev->id.vendor == 0 && dev->id.product == 0) {
+		printk (KERN_WARNING "mousedev ignoring device: %s\n", dev->name);
+		return 0;
+	}
+	
 	for (minor = 0; minor < MOUSEDEV_MINORS; minor++)
 		if (!mousedev_table[minor])
 			break;
-- 
1.7.4.1

